matlab/test_image_reconstruction.m

```diff
@@ -236,8 +238,20 @@
         mask = trace_map > 0.05; % reconstruct only gradients with small covariance
         grad_map_clip.x(mask) = 0;
         grad_map_clip.y(mask) = 0;
-        div = divergence(grad_map_clip.x,grad_map_clip.y);
-        rec_image = reshape( poicalc(-div(:),1,1,pano_height,pano_width), pano_height,pano_width);
+
+        %The lines below are different reconstruction methods
+        
+        %Method 1: (original) Requires the Partial Differential Equation toolbox
+        %div = divergence(grad_map_clip.x,grad_map_clip.y);
+        %rec_image = reshape( poicalc(-div(:),1,1,pano_height,pano_width), pano_height,pano_width);
+        
+        %Method 2: Requires the Image Processing toolbox
+        %the code is from http://www.cs.cmu.edu/~ILIM/projects/IM/aagrawal/software.html
+        %rec_image = poisson_solver_function_neumann(grad_map_clip.x, grad_map_clip.y);
+        
+        %Method 3: Does not require a toolbox.
+        %the code is from http://www.cs.cmu.edu/~ILIM/projects/IM/aagrawal/software.html
+        rec_image = frankotchellappa(grad_map_clip.x, grad_map_clip.y);
```

Comment:
It would be nice to comment on the differences between the methods. 
I carried out a simple comparison, using the same input, the gradient in variable `grad_map_clip`, and calling the three methods to produce the reconstructed images.
 
Before comparing the reconstructed images, I standardized them (zero mean and unit variance) since they did not have the same magnitude. In the following, I display all images in the same range. Here are the results:

**Reconstructed images**:
- Method 1, `poicalc`, uses Dirichlet boundary conditions (b.c.):
![screenshot from 2017-07-24 12-56-41](https://user-images.githubusercontent.com/8024432/28520501-04286d56-7070-11e7-9fe0-bcb73cac42a9.png)

- Method 2, `poisson_solver_function_neumann`, uses Neumann b.c.:
![screenshot from 2017-07-24 12-56-37](https://user-images.githubusercontent.com/8024432/28520526-294fb404-7070-11e7-94f9-2e9b95221cbb.png)

- Method 3, `frankotchellappa`, uses Neumann b.c.:
![screenshot from 2017-07-24 12-56-31](https://user-images.githubusercontent.com/8024432/28520542-39f8d934-7070-11e7-8bb2-dc7b65fc8424.png)

**Difference images**:
- 1 vs 3: they only differ near the boundary of the image. Good, since they use different b.c. (Dirichlet vs. Neumann).
![screenshot from 2017-07-24 12-56-26](https://user-images.githubusercontent.com/8024432/28520562-4d63619c-7070-11e7-8841-f017016efe5c.png)

- 1 vs. 2: they differ a lot more than just at the boundary of the image
![screenshot from 2017-07-24 12-56-22](https://user-images.githubusercontent.com/8024432/28520601-7909f57c-7070-11e7-97cf-8f0fe4168ad2.png)

- 2 vs. 3: even though the use the same b.c. (Neumann), they do not produce the same results. 
Method 2 seems to produce an image with smaller contrast than method 3.
![screenshot from 2017-07-24 12-56-16](https://user-images.githubusercontent.com/8024432/28520626-8dc1e1aa-7070-11e7-893c-74c8973ab67a.png)